### PR TITLE
fix windows build: nodejs uses posix style paths

### PR DESCRIPTION
**Pull request by samouri**
**summary**
I noticed that the Windows build has [been failing](https://github.com/ampproject/amphtml/runs/5009615400?check_suite_focus=true) for the past week. The failure was introduced by https://github.com/ampproject/amphtml/pull/36639. Nodejs expects posix style paths for `require` statements.

### CODE REVIEW

File: .github/workflows/cross-platform-builds.yml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 name: Cross-Platform Builds
 
-on:
-  push:
-    branches:
-      - main
+on: [pull_request]
```
Comment (rsimha) [inline .github/workflows/cross-platform-builds.yml:3]: This change did cause the workflow to run during the PR check, but it didn't actually do anything. To fully test, you'd also need to modify line 20 here as follows. (The last param is `prBuildWorkflow`, you can simply pass in `pushBuildWorkflow` as an arg for now.)

https://github.com/ampproject/amphtml/blob/07cb943d06ae2584696ffea6d2ef457fffde7838/build-system/pr-check/cross-platform-builds.js#L13-L20

```js
runCiJob(jobName, pushBuildWorkflow, pushBuildWorkflow)
```

File: build-system/tasks/extension-helpers.js [hunk 2]
```diff
@@ -746,7 +746,7 @@ async function getBentoBuildFilename(dir, name, mode, options) {
  * @return {string}
  */
 function generateBentoEntryPointSource(name, toExport, outputFilename) {
-  const bentoCePath = path.relative(
+  const bentoCePath = path.posix.relative(
```
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:749]: Looks like this will work. Approving in advance so you can merge once you're satisfied.

**Review by rsimha — APPROVED** [review]

File: build-system/tasks/extension-helpers.js (hunk 2 above)
Comment (rsimha) [reply build-system/tasks/extension-helpers.js:749]: On second thought, it won't. Imports need to be relative paths, so this should probably just be something like:
```js
const bentoCePath = '../../src/bento-ce'
```

Comment (samouri) [reply build-system/tasks/extension-helpers.js:749]: `path.posix.relative` returns a relative path. Should work now

File: .github/workflows/cross-platform-builds.yml (hunk 1 above)
Comment (samouri) [reply .github/workflows/cross-platform-builds.yml:3]: Thank you for the info and for catching this! You were right of course

**Review by rileyajones — APPROVED** [review]
Thanks for catching this. Windows pathing never fails to disappoint me.

**Pull request merged**